### PR TITLE
[2.3] Remove most refs uses

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13813 |
| License | MIT |
| Doc PR | - |

Removes some references usages. They are error prone, and trigger too many copies-on-writes.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php [hunk 1]
```diff
@@ -117,8 +117,10 @@ private function sanitizeQueries($connectionName, $queries)
     private function sanitizeQuery($connectionName, $query)
     {
         $query['explainable'] = true;
-        $query['params'] = (array) $query['params'];
-        foreach ($query['params'] as $j => &$param) {
+        if (!is_array($query['params'])) {
+            $query['params'] = array($query['params']);
+        }
```
Comment (WouterJ) [inline src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php]: I think `is_array($x) ? $x : array($x)` is more commonly used in the Symfony code base for these cases

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php [hunk 2]
```diff
@@ -36,13 +36,13 @@ public function load(array $configs, ContainerBuilder $container)
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('twig.xml');
 
-        foreach ($configs as &$config) {
+        foreach ($configs as $key => $config) {
```
Comment (WouterJ) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: isn't `array_map` better to use here?

File: src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php]: Both the array cast and the ternary operators trigger memory copies that are prevented with this notation

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php (hunk 2 above)
Comment (nicolas-grekas) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: not sure why...

File: src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php (hunk 1 above)
Comment (KingCrunch) [inline src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php]: Actually


```php
$query['params'] = (array) $query['params'];
```

Works too and imo is the most compact notation

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**